src/main/g8/scalafix/build.sbt

```diff
@@ -14,29 +18,45 @@ inThisBuild(
         url("https://example.com")
       )
     ),
-    addCompilerPlugin(scalafixSemanticdb),
-    scalacOptions ++= List(
-      "-Yrangepos",
-      "-P:semanticdb:synthetics:on"
-    )
+    semanticdbEnabled := true,
+    semanticdbIncludeInJar := true,
```

Comment:
This is required to make tests pass, but it shouldn't be. Removing it yields
```
[info]   scalafix.v1.SemanticDocument$Error$MissingSemanticdb: SemanticDB not found: input/src/main/scala/fix/MyRepo.scala
```

Despite
```
sbt:scalafix> ++ 3.0.0
[info] Setting Scala version to 3.0.0 on 3 projects.
[info] Excluded 3 projects, run ++ 3.0.0 -v for more details.
[info] Reapplying settings...
[info] set current project to scalafix (in build file:/home/bjaglin/git/projects/scalafix.g8/target/sbt-test/scalafix.g8-root/scripted/scalafix/)
sbt:scalafix> show tests/scalafixTestkitInputScalacOptions
[info] * -Xsemanticdb
[info] * -semanticdb-target
[info] * /home/bjaglin/git/projects/scalafix.g8/target/sbt-test/scalafix.g8-root/scripted/scalafix/input/target/scala-3.0.0/meta
```
```
sbt:scalafix> ++ 2.13.5
[info] Setting Scala version to 2.13.5 on 4 projects.
[info] Excluded 2 projects, run ++ 2.13.5 -v for more details.
[info] Reapplying settings...
[info] set current project to scalafix (in build file:/home/bjaglin/git/projects/scalafix.g8/target/sbt-test/scalafix.g8-root/scripted/scalafix/)
sbt:scalafix> show tests/scalafixTestkitInputScalacOptions
[info] * -P:semanticdb:synthetics:on
[info] * -Xplugin:/home/bjaglin/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/semanticdb-scalac_2.13.5/4.4.15/semanticdb-scalac_2.13.5-4.4.15.jar
[info] * -Yrangepos
[info] * -P:semanticdb:targetroot:/home/bjaglin/git/projects/scalafix.g8/target/sbt-test/scalafix.g8-root/scripted/scalafix/input/target/scala-2.13/meta

```
```
$ find . | grep MyRepo.scala.semanticdb
./input/target/scala-3.0.0/meta/META-INF/semanticdb/input/src/main/scala/fix/MyRepo.scala.semanticdb
./input/target/scala-2.12/meta/META-INF/semanticdb/input/src/main/scala/fix/MyRepo.scala.semanticdb
./input/target/scala-2.11/meta/META-INF/semanticdb/input/src/main/scala/fix/MyRepo.scala.semanticdb
./input/target/scala-2.13/meta/META-INF/semanticdb/input/src/main/scala/fix/MyRepo.scala.semanticdb

```

Reply:
https://github.com/scalacenter/scalafix/pull/1392/files#diff-3732d353a5ef3b44d0ae648cce3bc8fdf0c3c9c666a0150fa455879f3b9e662eR34 doesn't look right, but it looks like another problem.

Reply:
It's intersting! I struggled with semanticdb many times theses weeks (when I was developping for example the migration rule), and I had to add semanticdbInJar, I wasnt using the snapshot version, but the latest. 
I wonder if using the new api would make the detection of semanticdb files easier and clearer.

Reply:
I will try to spend some time on this. Just to understand if it's something new or not.

Reply:
This is only a problem with Scala 3 (the flag is parsed properly for Scala 2 before/after this PR), so the problem is either in the Scala 3 flag parsing (but the unit test looks legit), or the way `ScalafixTestKitPlugin` / `scalafix-testkit` pass the `scalaVersion` (I suspect it defaults to the runtime of the testkit, which is not 3.0.0). I will look at it again tonight as well, since I feel like it's the last part where Scala 3 support is not proper.

Reply:
> the flag is parsed properly for Scala 2 before/after this PR

Before, it was/is tested in https://github.com/scalacenter/sbt-scalafix/commit/4278a1f4f5e7ff28d086f8431c7e64af6f5d4084 and now also in https://github.com/scalacenter/scalafix.g8/pull/19/commits/22f6e3bf345ff258a0151d73770e5f41f2fefb3e#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91R13.

Reply:
There we are: https://github.com/scalacenter/sbt-scalafix/commit/cd9efb1b1e16f178da6be72a4cc621524e8a301a#diff-9db6645e9a45c4ebf9cc4085044d892745263231d8dc14af72e1933f4a538790 - the logic is repeated there! I will check tonight (unless you beat me to it) why this is not in `scalafix-testkit` since `scalacOptions` are passed also when using the testkit...